django/test/runner.py

```diff
@@ -615,6 +615,31 @@ def is_discoverable(label):
     return os.path.isdir(os.path.abspath(label))
 
 
+def reorder_preprocess(bin_i, test_class):
+    if test_class == TransactionTestCase and len(bin_i):
```

Comment:
We need to add this `_next_serialized_rollback` only on `TransactionTestCase` tests, and we have to ensure that there is not a mix between different test case classes.

It's specially important for the last `TransactionTestCase` which should have this parameter set to `True`. That's why the `TransactionTestCase` should also be grouped explicitly in the `DiscoveryRunner`.

If you agree with this idea, I need to add tests for it, and update documentation about test ordering.